tools/lib/NQP/Config/NQP.pm

```diff
@@ -331,7 +331,7 @@ sub gen_moar {
     my $force_rebuild = $options->{'force-rebuild'};
     my $has_gen_moar  = defined $gen_moar;
     my @opts          = @{ $options->{'moar-option'} || [] };
-    push @opts, "--optimize";
+    push @opts, "--optimize" unless grep('--no-optimize', @{ $options->{'moar-option'} });
```

Comment:
I don't think, this works as intended. It will still always set --optimize:
```
> perl -E 'say grep("d", qw(b a c));'
bac
> perl -E 'say grep({$_ eq "d"} qw(b a c));'

> perl -E 'say grep({$_ eq "a"} qw(b a c));'
a
```